### ISSUE TITLE
cuLinkCreate CUDA_ERROR_INVALID_VALUE on sm_89/driver 590.48 (CUDA 13.1)

### ISSUE THREAD

**Issue by noahgift** [paiml/trueno#231]
## Summary

`cuLinkCreate` fails with `CUDA_ERROR_INVALID_VALUE` on RTX 4060L (sm_89) with CUDA 13.1 driver 590.48.01. This causes all PTX kernel compilation via the linker API to fail, which cascades into entrenar training forward falling back to CPU (186x throughput deficit).

## Evidence

From `qwen-train-canary` canary runs on yoga:
```
[PTX-CACHE] Linker compilation failed: CUDA module loading failed: 
cuLinkCreate failed: CUDA driver error: CUDA_ERROR_INVALID_VALUE (code: 1), 
falling through to legacy JIT
```

This message repeats for every kernel load (~30 kernels).

## Root Cause

In `trueno-gpu/src/driver/module.rs:88-97`:
```rust
let result = if use_jit_target {
    unsafe {
        (driver.cuLinkCreate)(1, opt_keys.as_mut_ptr(), opt_vals.as_mut_ptr(), &mut link_state)
    }
} else {
    unsafe { (driver.cuLinkCreate)(0, ptr::null_mut(), ptr::null_mut(), &mut link_state) }
};
```

The `CU_JIT_TARGET=89` option passed to `cuLinkCreate` returns `CUDA_ERROR_INVALID_VALUE` on this driver version. The legacy fallback (`cuModuleLoadDataEx`) works, but the training CUDA forward still fails separately.

## Impact

- **186x throughput deficit** in apr finetune (36 tok/s vs unsloth 6,697 tok/s)
- Affects yoga (sm_89) with driver 590.48.01
- Does NOT affect gx10 (sm_121) which has a different failure mode (NaN/Inf logits)

## Contract

Provable contract: `qwen-train-canary/contracts/cuda-training-forward-v1.yaml`
Falsification test: FALSIFY-TRAIN-004

## Refs

- PMAT-420 (qwen-train-canary)
- F-WL-06 in training canary spec

**Comment by noahgift** [comment]
**Triage: P1 — GPU/CUDA correctness.**

Hardware-specific GPU issue. These are blocking for CUDA training/inference pipelines on affected architectures. Grouped with other GPU issues (#231, #226, #207, #203, #200, #184, #180, #170) for batch resolution during next CUDA stabilization sprint.
